<a href="https://colab.research.google.com/github/reshmapasula-prog/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/reshmapasula-prog/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [14]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


Refresh / Content Opportunity Scoring

I want to identify pages that may need a refresh and rank them by opportunity. The goal is to help a content team decide which pages should be reviewed first. This lane is useful because the starter data contains observable search and content signals such as impressions, clicks, CTR, position, content age, word count, and trend direction. I will use these signals to find patterns and create evidence-based recommendations rather than simply training a model.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [15]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


Research question: Which pages should a content team review first for a possible refresh?

Decision: Prioritize pages that show signs of declining search performance or a meaningful opportunity for improvement.

Action: A content team member reviews the highest-priority pages and decides whether to update them.

Cost of a wrong recommendation: If a page is incorrectly prioritized, the team may waste time updating a page that does not need improvement, while a genuinely declining page may be missed.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [16]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

data_path = "/content/flyrank-ml-internship-starter/data/raw/content_refresh_anonymized.csv"

df = pd.read_csv(data_path)

print("Rows:", len(df))
print("Columns:", len(df.columns))
print("Median impressions:", df["impressions_90d"].median())
print("Median CTR:", df["ctr"].median())
print("Declining pages:", (df["trend_direction"] == "down").mean())

Rows: 30000
Columns: 44
Median impressions: 731.0
Median CTR: 0.07
Declining pages: 0.5420666666666667


The starter dataset contains 30,000 pages and 44 columns. The median page has 731 impressions over 90 days, and the median CTR is 0.07. About 54.2% of pages have a downward trend. These numbers suggest there is enough search-performance and content data to investigate which pages may be worth reviewing for a refresh. These are observed patterns, not proof that a refresh will improve performance.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [17]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


This work can identify observed patterns and directional relationships in the available search and content data. It can provide decision-support for prioritizing pages for review. However, it cannot prove that one factor caused a change in search performance, and it cannot claim to predict Google's algorithm or guarantee that a recommended page will improve after a refresh. The results should therefore be treated as evidence for prioritization, not as causal proof.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.